spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/reactive/RetryableLoadBalancerExchangeFilterFunction.java

```diff
@@ -229,10 +229,15 @@ private boolean shouldRetryNextServiceInstance(LoadBalancerRetryPolicy retryPoli
 	}
 
 	private boolean isRetryException(Throwable throwable) {
-		return exceptions.stream()
-				.anyMatch(exception -> exception.isInstance(throwable)
-						|| throwable != null && exception.isInstance(throwable.getCause())
-						|| Exceptions.isRetryExhausted(throwable));
+		if (throwable == null) {
+			return false;
+		}
+		return isInstanceOfRetryType(throwable) || isInstanceOfRetryType(throwable.getCause())
+				|| Exceptions.isRetryExhausted(throwable);
+	}
+
+	private boolean isInstanceOfRetryType(Throwable throwable) {
+		return typesThatTriggerRetry.stream().anyMatch(type -> type.isInstance(throwable));
```

Comment:
I would like to replace it with 
```suggestion
		return throwable instanceof IOException
		        ||  throwable instanceof TimeoutException
		        ||  throwable instanceof RetryableStatusCodeException;
```
because it is simpler and it moves the information about the types close to where they are used. For me that makes the code easier to read. What do you think?